src/util/db/sqlqueryfinisher.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MIXXX_SQLQUERYFINISHER_H
+#define MIXXX_SQLQUERYFINISHER_H
+
+
+#include <QSqlQuery>
+
+
+// Ensures that reusable queries are properly finished when
+// leaving the corresponding execution scope. This will free
+// resources of prepared statements until the next execution.
+class SqlQueryFinisher final {
+public:
+    explicit SqlQueryFinisher(QSqlQuery query)
+        : m_query(query) {
+    }
+    SqlQueryFinisher(SqlQueryFinisher&& other)
+        : m_query(std::move(other.m_query)) {
```

Comment:
It doesn't matter if the target class does not provide move semantics. Moving has stronger preconditions, but weaker postconditions than copying. The strong preconditions are matched by the r-value reference. By using using std::move() I'm just declaring that I will not access the argument afterwards and that it's contents "might" be moved. If they are actually copied, fine, I don't care, because I never ever will access other.m_query again. If the other class ever implements a more efficient move constructor and I don't use std::move() here the less efficient copy constructor will be used forever. Just because I coded a copy although I don't need a 1:1 copy!

I begin to hate C++ while studying Rust ;) Rust has all those concepts integrated into the language itself.